src/server/logging/applyFilterToKey.js

```diff
@@ -0,0 +1,46 @@
+function toPojo(obj) {
+  return JSON.parse(JSON.stringify(obj));
+}
+
+function replacer(match, group) {
+  return (new Array(group.length + 1).join('X'));
+}
+
+function apply(obj, key, action) {
+  for (let k in obj)  {
+    if (obj.hasOwnProperty(k)) {
+      let val = obj[k];
+      if (k === key) {
+        if (action === 'remove') {
+          delete obj[k];
+        }
+        else if (action === 'censor' && typeof val === 'object') {
+          delete obj[key];
+        }
+        else if (action === 'censor') {
+          obj[k] = ('' + val).replace(/./g, 'X');
+        }
+        else if (/\/.+\//.test(action)) {
+          var matches = action.match(/\/(.+)\//);
+          if (matches) {
+            let regex = new RegExp(matches[1]);
+            obj[k] = ('' + val).replace(regex, replacer);
+          }
+        }
+      } else if (typeof val === 'object') {
+        val = apply(val, key, action);
+      }
+    }
+  }
+  return obj;
+}
+
+module.exports = function applyFilterToKey(obj, key, action) {
```

Comment:
This honestly seems a tad janky to me to have one function exported by `module.exports` and one as a property (`module.exports. applyFiltersToKeys`). I think personally it would be better to export just `applyFiltersToKeys`, seeing as how that's all that is used anyway (besides in the tests), or export both under keys (`applyFiltersToKey` and `applyFiltersToKeys`).